tests/measurements/test_var.py

```diff
@@ -19,13 +19,40 @@
 from pennylane.measurements import Variance
 
 
+# TODO: Remove this when new CustomMP are the default
+def custom_measurement_process(device, spy):
+    assert len(spy.call_args_list) > 0  # make sure method is mocked properly
+
+    samples = device._samples
+    state = device._state
+    call_args_list = list(spy.call_args_list)
+    for call_args in call_args_list:
+        obs = call_args.args[1]
+        shot_range, bin_size = (
+            call_args.kwargs["shot_range"],
+            call_args.kwargs["bin_size"],
+        )
+        meas = qml.var(op=obs)
+        old_res = device.var(obs, shot_range, bin_size)
+        if samples is not None:
+            new_res = meas.process_samples(
+                samples=samples, wire_order=device.wires, shot_range=shot_range, bin_size=bin_size
+            )
+        else:
+            new_res = meas.process_state(state=state, wire_order=device.wires)
+        assert qml.math.allequal(old_res, new_res)
```

Comment:
I am not too familiar with mocker, can you explain what's happening here with spy? Specifically, in the tests you define spy as : `spy = mocker.spy(qml.QubitDevice, "var")`. This makes me think spy will keep track of the call arguments passed to the `var()` method of the `QubitDevice` class. But I am confused as to how that will even be called if in the circuit we are calling `qml.var` which routes to your new class.

Reply:
In `QubitDevice` we are still calling `self.var` instead of `_Variance.process_samples`/ `process_state`, thus these new methods are never tested. In here I basically track all the calls to `QubitDevice.var`, I use the args to do the exact same calls to `process_samples` and `process_states` and then compare the results.

Reply:
but in the circuit you return `qml.var(op=obs)`, doesn't this return and instance of your new class?

Reply:
Yep, but in `QubitDevice` we check for the `return_type` of the measurement, which hasn't changed.

Reply:
It does, but the new class inherits from `MeasurementProcess` so the statistics are computed the same way as before. Currently `qml.var.process_samples` is not called.